extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java

```diff
@@ -92,6 +94,16 @@ public void handleMessage(Object msg) {
                     HttpResponse res = (HttpResponse) msg;
                     responseBuilder.setStatusCode(res.status().code());
 
+                    final HttpHeaders responseHeaders = res.headers();
+                    if (responseHeaders != null) {
+                        final String contentType = responseHeaders.getAsString(HttpHeaderNames.CONTENT_TYPE);
+                        if (contentType != null && contentType.trim().length() > 0) {
+                            final Map<String, String> headers = new HashMap<>();
+                            headers.put("Content-Type", contentType);
+                            responseBuilder.setHeaders(headers);
+                        }
+                    }
+
                     Headers multiValueHeaders = new Headers();
                     responseBuilder.setMultiValueHeaders(multiValueHeaders);
                     for (String name : res.headers().names()) {
```

Comment:
I saw that, but without my change the Lambda is returning 'text/plain' rather than 'application/json'. Digging a bit further I found ["Format 2.0 doesn't have multiValueHeaders or multiValueQueryStringParameters fields"](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html) on the AWS API Gateway documentation.

(as stated in #15909 I'm using API Gateway v2)

If you'd rather I used your approach to copy across all headers, but use the setHeaders() method instead, I can easily do this.

Reply:
Richie,

The code uses com.amazonaws.services.lambda.runtime.events.APIGatewayV2HTTPResponse which does have a multivalueheaders property.   I guess when they run within Lambda runtime without our wrapper they marshal this correctly?  Oh well, your fix looks good.